extensions/wikia/Wall/WallThread.class.php

```diff
@@ -109,13 +113,25 @@ private function loadReplyIdsFromDB($master = false) {
 			$list[] = $row->comment_id;
 		}
 
-		$this->setReplies( $list );
+		$lastId = end( $list );
+
+		return empty( $lastId ) ? $list :
+			array_merge( $list, $this->getReplyIdsFromDB( $dbr, $lastId ) );
+	}
+
+	private function loadReplyIdsFromDB( $master = false ) {
+		if ( empty( Title::newFromId( $this->mThreadId ) ) ) {
+			return;
+		}
+
+		$dbr = wfGetDB( $master ? DB_MASTER : DB_SLAVE );
+
+		$this->setReplies( $this->getReplyIdsFromDB( $dbr ) );
 	}
 
 	public function invalidateCache() {
 		// invalidate cache at Thread level (new reply or reply removed in thread)
-		$this->mForceMaster = true;
-		$this->loadReplyIdsFromDB( true );
+		$this->getCache()->delete( $this->getThreadKey() );
```

Comment:
Calling ```loadReplyIdsFromDB``` would eventually call ```setReplies``` which would do some initialization on ```data```:
```php
$this->data = new StdClass();
$this->data->threadReplyIds = false;
$this->data->threadReplyObjs = false;
```
We should probably do that here just to make sure the local copy of this now out of date data is removed.